terraform/data_upload_lambda.tf

```diff
@@ -0,0 +1,122 @@
+# IAM Role for Lambda Execution
+resource "aws_iam_role" "c14_lambda_execution_role" {
+  name = "c14-earthquake-monitor-etl-lambda_execution_role"
+
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Action    = "sts:AssumeRole"
+        Effect    = "Allow"
+        Principal = {
+          Service = "lambda.amazonaws.com"
+        }
+      }
+    ]
+  })
+}
+
+# IAM Policy for RDS and S3 Access
+resource "aws_iam_role_policy" "c14_lambda_execution_policy" {
+  name = "c14-earthquake-monitor-etl-lambda_execution_policy"
+  role = aws_iam_role.c14_lambda_execution_role.id
+
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Action = [
+          "logs:CreateLogGroup",
+          "logs:CreateLogStream",
+          "logs:PutLogEvents"
+        ]
+        Effect   = "Allow"
+        Resource = "arn:aws:logs:*:*:*"
+      },
+      {
+        Action   = ["rds:DescribeDBInstances", "rds-db:connect"],
+        Effect   = "Allow"
+        Resource = "arn:aws:rds-db:eu-west-2:${var.ACCOUNT_ID}:dbuser:${var.RDS_RESOURCE_ID}/${var.DB_USER}"
+      },
+      {
+        Action   = ["s3:PutObject", "s3:GetObject"],
+        Effect   = "Allow"
+        Resource = "arn:aws:s3:::${var.S3_BUCKET}/*"
+      }
+    ]
+  })
+}
+
+# Lambda Function using Docker Image
+resource "aws_lambda_function" "c14_lambda_function" {
+  function_name    = "c14-earthquake-monitor-etl-lambda"
+  role             = aws_iam_role.c14_lambda_execution_role.arn
+  package_type     = "Image"
+  image_uri        = "${var.ETL_ECR_URI}:latest"
```

Comment:
You might have to change this as the latest ECR image is usually an image index that doesn't work with lambdas.

Reply:
Ok all good